lib/filterx/tests/test_expr_comparison.c

```diff
@@ -570,29 +576,29 @@ Test(expr_comparison, test_string_cases_type_aware_comparison)
   // check if compared as string
   _assert_comparison(filterx_string_new("a443", 4), filterx_integer_new(443), FCMPX_EQ | FCMPX_TYPE_AWARE, FALSE);
   _assert_comparison(filterx_string_new("a443", 4), filterx_integer_new(443), FCMPX_LT | FCMPX_TYPE_AWARE, FALSE);
-  _assert_comparison(filterx_string_new("a443", 4), filterx_integer_new(443), FCMPX_GT | FCMPX_TYPE_AWARE, TRUE);
+  _assert_comparison(filterx_string_new("a443", 4), filterx_integer_new(443), FCMPX_GT | FCMPX_TYPE_AWARE, FALSE);
   _assert_comparison(filterx_string_new("a443", 4), filterx_integer_new(443), FCMPX_NE  | FCMPX_TYPE_AWARE, TRUE);
 
   // bytes - boolean
-  _assert_comparison(filterx_bytes_new("true", 4), filterx_boolean_new(true), FCMPX_EQ | FCMPX_TYPE_AWARE, TRUE);
+  _assert_comparison(filterx_bytes_new("true", 4), filterx_boolean_new(true), FCMPX_EQ | FCMPX_TYPE_AWARE, FALSE);
```

Comment:
This would have been great, but I guess it's not a big deal as it works with strings.

Reply:
We can write custom parsing logic for these, and convert them to 0 or 1 numbers then follow through with the number comparison. It might make sense.